mysite/static/js/search/apiWrapper.js

```diff
@@ -0,0 +1,643 @@
+/**
+apiWrapper.js
+
+The Objects in this file represent wrapper around various bug tracker APIs.
+When possible, these wrappers inherit from a parent. They all inherit
+from APIWrapper. Individual wrappers like JiraMifos will inherit
+from the more general Mifos wrapper. This allows additional wrappers
+to be added for individual cases easily. You can override any attribute
+or method in a child by simply writing the method yourself in that class.
+
+All Wrapper objects must contain - or inherit - an executeAPI() function
+as well as a populateModal() function. The first of these two is the first
+and perhaps last set of function that makes the API request (callback should
+always be used in asynchronous situations). populateModal() simply
+fills in the popup modal with data recieved by the API request.
+
+Important distinctions made in this file - tracker_type, tracker_id.
+The tracker_type is one level of generalization over tracker_id. For
+example, tracker_type can be an integer that refers to the Launchpad
+bug tracker. However, each instance of Launchpad (heat, horizon, etc)
+will recieve their own tracker _id. 
+
+IMPORTANT: Make sure you have CORS enabled if you are developing locally.
+*/
+
+/**
+	TODO: 
+		1) Finish Python BugTracker
+		2) Make timeout on spinny wheel thing
+			Does that mean timeout on request too?
+			Fill the modal with a different message
+*/
+
+
+/**
+DataCache is a local (not currently using cookies) record of which APIs
```

Comment:
In the long run, if you want to store data locally, consider using `localStorage` rather than cookies, for what it's worth.

http://people.rit.edu/tlh4225/736/exercises/storageWriteup.html explains the differences.